go/pkg/cas/upload.go

```diff
@@ -68,8 +76,24 @@ type UploadOptions struct {
 	//
 	// This field is ignored if PreserveSymlinks is false, which is the default.
 	AllowDanglingSymlinks bool
+
+	// Callback is called for each file/dir to be uploaded.
+	// If it returns an error which is ErrSkip according to errors.Is, then the
+	// file/dir is not not uploaded.
+	// If it returns another error, then the upload is halted with that error.
+	//
+	// Callback is called from different goroutines.
+	Callback func(absPath string, mode os.FileMode) error
```

Comment:
We should document that the Callback is not reflected at on the cache.

Particularly, to allow for skipping processing files, Callback can be called repeatedly for the same file, and the files processed by callback might still not be included/uploaded since this happens before processing the file.

Reply:
that's a great observation. Thank you!

I've changed the order of Callback/regexp checks, so now Callback func doesn't have to bother with PathExclude.

I've documented that Callback might be called twice for the same file.